containerd-shim-spin-v1/src/main.rs

```diff
@@ -183,12 +169,64 @@ impl Instance for Wasi {
-                        tls_cert: None,
-                        tls_key: None,
-                    });
+                    let trigger = app.info.trigger.clone();
+                    info!(" >>> building spin trigger {:?}", trigger);
+
+                    let f: Pin<Box<dyn Future<Output = Result<(), anyhow::Error>> + Send>>;
+
+                    match trigger {
+                        spin_manifest::ApplicationTrigger::Http(_config) => {
+                            let http_trigger: HttpTrigger = match Wasi::build_spin_trigger(
+                                working_dir,
+                                app,
+                                PathBuf::from(stdout),
+                                PathBuf::from(stderr),
+                                PathBuf::from(stdin),
+                            )
+                            .await
+                            {
+                                Ok(http_trigger) => http_trigger,
+                                Err(err) => {
+                                    tx.send(Err(Error::Others(format!(
+                                        "could not build spin trigger: {err}"
+                                    ))))
+                                    .unwrap();
+                                    return;
+                                }
+                            };
+
+                            info!(" >>> running spin trigger");
+                            f = http_trigger.run(spin_trigger_http::CliArgs {
+                                address: parse_addr(SPIN_ADDR).unwrap(),
+                                tls_cert: None,
+                                tls_key: None,
+                            });
+                        }
+                        spin_manifest::ApplicationTrigger::Redis(_config) => {
+                            let redis_trigger: RedisTrigger = match Wasi::build_spin_trigger(
+                                working_dir,
+                                app,
+                                PathBuf::from(stdout),
+                                PathBuf::from(stderr),
+                                PathBuf::from(stdin),
+                            )
+                            .await
+                            {
+                                Ok(redis_trigger) => redis_trigger,
+                                Err(err) => {
+                                    tx.send(Err(Error::Others(format!(
+                                        "could not build spin trigger: {err}"
+                                    ))))
+                                    .unwrap();
+                                    return;
+                                }
+                            };
+
+                            info!(" >>> running spin trigger");
+                            f = redis_trigger.run(spin_trigger::cli::NoArgs);
+                        }
+                        _ => todo!("Only Http and Redis triggers are currently supported."),
```

Comment:
I am working on a generic trigger implementaion for the spin shim. This definitely helps!